File: Serene/Serene.Script/Imports/ServiceContracts/Administration.LanguageRow.cs

namespace Serene.Administration
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class LanguageRow
    {
        [InlineConstant] public const string IdProperty = "Id";
        [InlineConstant] public const string NameProperty = "LanguageName";
        [InlineConstant] public const string LocalTextPrefix = "Administration.Language";
        [InlineConstant] public const string LookupKey = "Administration.Language";
    
        public static Lookup<LanguageRow> Lookup { [InlineCode("Q.getLookup('Administration.Language')")] get { return null; } }
    
        public Int32? Id { get; set; }
        public String LanguageId { get; set; }
        public String LanguageName { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string Id = "Id";
            [InlineConstant] public const string LanguageId = "LanguageId";
            [InlineConstant] public const string LanguageName = "LanguageName";
        }
    }
    
}

File: Serene/Serene.Core/Imports/ServerTypings/Administration.LanguageRow.ts
namespace Serene.Administration {
    export interface LanguageRow {
        Id?: number;
        LanguageId?: string;
        LanguageName?: string;
    }

    export namespace LanguageRow {
        export const idProperty = 'Id';
        export const nameProperty = 'LanguageName';
        export const localTextPrefix = 'Administration.Language';
        export const lookupKey = 'Administration.Language';

        export function getLookup(): Q.Lookup<LanguageRow> {
            return Q.getLookup<LanguageRow>('Administration.Language');
        }

        export namespace Fields {
            export declare const Id: string;
            export declare const LanguageId: string;
            export declare const LanguageName: string;
        }

        [
            'Id', 
            'LanguageId', 
            'LanguageName'
        ].forEach(x => (<any>Fields)[x] = x);
    }
}
